File: samples/ruby/items/items.rb
# retrieve all items
require 'net/http'
require 'json'

API_VERSION = 1
API_HOST    = 'https://api.thebase.in'
token       = '' # Retrieve token using ruby/oauth/token.rb

header_parameters =
  { 'Authorization' => "Bearer #{token}" }

request_parameters =
  { 'order'  => 'created',
    'sort'   => 'desc',
    'limit'  => '10',
    'offset' => '0' }

uri =
  URI([
    API_HOST,
    "/#{API_VERSION}/items/?",
    URI.encode_www_form(request_parameters)]
      .join)

request =
  Net::HTTP::Get.new(
    uri,
    header_parameters)

response =
  Net::HTTP.start(
    uri.hostname,
    uri.port,
    use_ssl: uri.scheme == 'https') do |http|
      http.request(request)
    end

parsed_response =
  JSON.parse response.body

File: samples/php/items/items.php
<?php
// 商品一覧を取得する
define('API_HOST',    'https://api.thebase.in');
define('API_VERSION', '1');
$token = ''; # YOUR ACCESS TOKEN

$params = http_build_query(
    array(
        'order'  => 'created',
        'sort'   => 'desc',
        'limit'  => '10',
        'offset' => '0',
    )
);
$headers = array(
    'Authorization: Bearer ' . $token,
);
$request_options = array(
    'http' => array(
        'method' => 'GET',
        'header' => implode("\r\n", $headers),
        'ignore_errors' => true,
    ),
);
$context = stream_context_create($request_options);

$response_body = file_get_contents(API_HOST . '/' . API_VERSION . '/items?' . $params, false, $context);
$parsed_response = json_decode($response_body, true);
